### PR TITLE
Blog - How to Set Up a Private Docker Registry on Linux

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
@balapriyac  [Preview Link](https://645a480303d3df1f64c21746--ephemeral-narwhal-3ac3b2.netlify.app/blog/private-docker-registry/)

File: blog/_posts/2023-06-13-private-docker-registry.md [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+snap install --classic certbot
+~~~
+
+Next, copy the snap binary to the system path.
+
+~~~{.bash caption=""}
+ln -s /snap/bin/certbot /usr/bin/certbot
+~~~
+
+Then, run the `certbot` command followed by your domain name to download the SSL certificates, and configure NGINX to use the downloaded certificates for your domain..
```
Comment (balapriyac) [inline blog/_posts/2023-06-13-private-docker-registry.md:306]:
```suggestion
Then, run the `certbot` command followed by your domain name to download the SSL certificates, and configure NGINX to use the downloaded certificates for your domain.
```

File: blog/_posts/2023-06-13-private-docker-registry.md [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+server_names_hash_bucket_size 64;
+~~~
+
+Restart the NGINX service to reload the changes.
+
+~~~{.bash caption=">_"}
+systemctl restart nginx
+~~~
+
+Run the following command to verify the NGINX running status
```
Comment (balapriyac) [inline blog/_posts/2023-06-13-private-docker-registry.md:276]:
```suggestion
Run the following command to verify the NGINX running status.
```

File: blog/_posts/2023-06-13-private-docker-registry.md [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
+    proxy_read_timeout 900;
+    }
+}
+~~~
+
+Save the file when you are done.
+
+Next, we'll need to define two more settings for NGINX..
```
Comment (balapriyac) [inline blog/_posts/2023-06-13-private-docker-registry.md:251]:
```suggestion
Next, we'll need to define two more settings for NGINX.
```

**Review by balapriyac — CHANGES_REQUESTED** [review]
@msali123 Great work, Sameer. Just suggested a couple of minor changes.

**Comment by msali123** [comment]
Thanks Bala [Changes done](https://6461ec64db1c4527db9b7c8b--ephemeral-narwhal-3ac3b2.netlify.app/blog/private-docker-registry/)

File: blog/_posts/2023-06-13-private-docker-registry.md [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+
+internal-links:
+ - Docker
+ - Registry
+ - Linux
+ - Security
+ - Container
+ - Images
+---
+
```
Comment (balapriyac) [inline blog/_posts/2023-06-13-private-docker-registry.md:16]:
```suggestion
**We're [Earthly.dev](https://earthly.dev/). We make building software simpler and therefore faster using containerization. This article covers setting up a private Docker registry on Linux. If you want to know more about building in containers then [check us out](/).**
```

**Review by balapriyac — CHANGES_REQUESTED** [review]
@msali123 added Earthly tie-in.

**Comment by msali123** [comment]
@balapriyac [Added Earthly tie-in](https://6463386fb9a39160cf76bd50--ephemeral-narwhal-3ac3b2.netlify.app/blog/private-docker-registry/)

**Comment by balapriyac** [comment]
@jalletto this is ready to merge.

**Pull request merged**